ntp-proto/src/timetypes.rs

```diff
@@ -298,4 +324,18 @@ mod tests {
     ntp_duration_scaling_test!(ntp_duration_scaling_u32, u32);
     ntp_duration_scaling_test!(ntp_duration_scaling_u64, u64);
     ntp_duration_scaling_test!(ntp_duration_scaling_usize, usize);
+
+    macro_rules! assert_eq_epsilon {
+        ($a:expr, $b:expr, $epsilon:expr) => {
+            assert!(($a - $b).abs() < $epsilon);
+        };
+    }
+
+    #[test]
+    fn duration_seconds_roundtrip() {
+        assert_eq_epsilon!(NtpDuration::from_seconds(0.0).to_seconds(), 0.0, 1e-9);
+        assert_eq_epsilon!(NtpDuration::from_seconds(1.0).to_seconds(), 1.0, 1e-9);
+        assert_eq_epsilon!(NtpDuration::from_seconds(1.5).to_seconds(), 1.5, 1e-9);
+        assert_eq_epsilon!(NtpDuration::from_seconds(2.0).to_seconds(), 2.0, 1e-9);
+    }
```

Comment:
we have to deal with precision loss here. We seem to be correct up to `1e-9` seconds, so nanoseconds. That seems acceptable to me